modules/groovy-dsl/Datasources.md

```diff
@@ -0,0 +1,168 @@
+# Datasources
+
+Datasources is likely the most complex feature in the Assert4J Test DSL, and unless you've already used DataSources in 
+[ReadyApi](https://smartbear.com/product/ready-api/overview/) or SoapUI Pro you may need some time to understand how to use them. 
+
+Arguably, it's also the most powerful feature in the DSL, however, so it's very likely that it will be worth the investment. 
+Datasources allow you to easily make your test data-driven and thus much more relevant and to generate test data if you don't have
+it. Read on to learn the basic concepts and syntax of DataSources.
+
+***NOTE:*** Datasources are only available for TestServer executions, not for local executions. In other words you need a ReadyApi TestServer 
+to be able to use them.
+
+## How Datasources work
+ 
+The idea of a Datasource is that it will loop over a number of data records. In every iteration, it will set one variable 
+per column in the datasource, and will pass these variables to a number of test steps.
+
+To insert the variables in your test you will then use SoapUI [property expansion](https://www.soapui.org/scripting---properties/property-expansion.html).
+
+For instance, assume that you have an Excel file called ```cities.xls``` with the following contents:
+
+|Country|City|
+|-------|-----|
+|France | Paris |
+|Sweden | Stockholm |
+|Italy | Rome |
+
+
+We're now going to use this data to execute the same test (two test steps) once for every row in the Excel sheet:
+
+```groovy
+ import static io.swagger.assert4j.dsl.execution.RecipeExecution.executeRecipe
+ 
+  executeRecipe {
```

Comment:
some of the examples use executeRecipe - some only show the content of that statement - should we be consistent on either?

Reply:
I'm not sure that's necessary. Sometimes it's better to be concise.